### ISSUE TITLE
Save shapelet model

### ISSUE THREAD

**Issue by gio8tisu** [tslearn-team/tslearn#53]
I would like to save my trained shapelet model. Any help will be appreciated.
Thanks.

**Comment by GillesVandewiele** [comment]
Depends on what you want to do:
* You can store the neural nets/optimizers that are discovering these shapelets. U can use the keras [save functionality](https://keras.io/getting-started/faq/#how-can-i-save-a-keras-model) for that. Could be useful if training could be interrupted somehow
* You can store the discovered shapelets. Access them with the [shapelets_](https://github.com/rtavenar/tslearn/blob/master/tslearn/shapelets.py#L197) attribute

**Comment by rtavenar** [comment]
Hi all,

Indeed, that would be a nice functionality. @gio8tisu  : if you just need to access your shapelets afterwards, I would get inspiration from the second solution given by @GillesVandewiele and do something like:

```python
from tslearn.utils import save_timeseries_txt

shapelets = my_trained_model.shapelets_as_time_series_
save_timeseries_txt("/path/to/my/file.txt", shapelets)
```

But definitely, it would be nice to be able to serialize the whole model. Hence, I add the "new feature" label (I really do not have time to do it right now, sorry).

NB : an in-between solution if you are used to `keras` would be to save all weights in separate files (you can access them via the `get_weights` method) and when you want to reuse your model, build a new shapelet model with the right architecture, reload all the weight files and set the model's weights accordingly.

**Comment by gio8tisu** [comment]
Thank you for your replies, I will try your suggestion for saving my learned shapelets, as I'm not really familiar with keras. Looking forward for this feature :smiley:
Thank you very much for your super helpful work.

**Comment by gio8tisu** [comment]
This seems to do the trick:
* In LocalSquaredDistanceLayer class I added:
```python
def get_config(self):
       return {'n_shapelets': self.n_shapelets}
```
* To load the model:
```python
load_model(filename, custom_objects={'GlobalMinPooling1D': GlobalMinPooling1D,
                                                               'LocalSquaredDistanceLayer': LocalSquaredDistanceLayer})
```

**Comment by GillesVandewiele** [comment]
Could you create a PR with a save function in the ShapeletModel, @gio8tisu? I'm sure @rtavenar would appreciate it :)

**Comment by murometz** [comment]
> Hi all,
> 
> Indeed, that would be a nice functionality. @gio8tisu : if you just need to access your shapelets afterwards, I would get inspiration from the second solution given by @GillesVandewiele and do something like:
> 
> ```python
> from tslearn.utils import save_timeseries_txt
> 
> shapelets = my_trained_model.shapelets_as_time_series_
> save_timeseries_txt("/path/to/my/file.txt", shapelets)
> ```
> 
> But definitely, it would be nice to be able to serialize the whole model. Hence, I add the "new feature" label (I really do not have time to do it right now, sorry).
> 
> NB : an in-between solution if you are used to `keras` would be to save all weights to disk (you can access them via the `get_weights` method) and when you want to reuse your model, build a new shapelet model with the correct architecture, reload all the weights and set the model's weights accordingly.

Hi. thank you very much for your work.

How to load weights?

I save the weights from trained model:

Create model:
shp_clf_1 = ShapeletModel(n_shapelets_per_size=shapelet_sizes,
                        optimizer=Adagrad(lr=.01),
                        weight_regularizer=.01,
                        max_iter=8000,
                        verbose_level=0)

Now I want to load weights and predict without training.

How?

Thank you very much!
Regards
Ilja Rasin

**Comment by rtavenar** [comment]
This is not implemented yet. See PR #57 for more info on that.

**Comment by rtavenar** [comment]
Implemented in #227
